### ISSUE TITLE
Apollo Ultra features not working

### ISSUE THREAD

**Issue by cameronwilliams** [JeffreyCA/Apollo-ImprovedCustomApi#1]
For Apollo 1.15.11, all Apollo Ultra features should be unlocked, but they’re not working.

**Comment by JeffreyCA** [comment]
Yes this is a known issue, any contributions/suggestions are welcome

**Comment by JeffreyCA** [comment]
Some findings:

| Feature | NSUserDefaults standardUserDefaults key | Value | Notes |
|---|---|---|--|
| Subreddit weather | `ShowSubredditWeatherTime` | `YES`/`NO` | Weather data doesn't load because of missing [WeatherKit entitlements]((https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_weatherkit)) | 
| New comment highlight | `ShowUnreadComments` | `YES`/`NO` | New comments aren't highlighted when set to `YES` |

**Comment by Proaxel** [comment]
I heard apps installed with Trollstore still have working entitlements, could Subreddit weather still work that way? I do have a Trollstore compatible device to test with, but it's pretty old at this point (iPhone 7 Plus on iOS 14.5). Either way, that's a bummer how it turned out to require those.

As for New Comments, perhaps there's some other thing that needs to be fulfilled...?

**Comment by cameronwilliams** [comment]
I don’t think it’s just entitlements. 1.15.11 should have all ultra features unlocked, but even the Pixel Pals don’t work.

**Comment by JeffreyCA** [comment]
> I heard apps installed with Trollstore still have working entitlements, could Subreddit weather still work that way? I do have a Trollstore compatible device to test with, but it's pretty old at this point (iPhone 7 Plus on iOS 14.5). Either way, that's a bummer how it turned out to require those.
> 
> As for New Comments, perhaps there's some other thing that needs to be fulfilled...?

I added some working toggles for **New Comments Highlightifier** and **Subreddit Weather** under the Custom API settings in [this experimental build](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/releases/tag/v1.0.6-experimental). Could you give it a test?

For me (without Trollstore), the weather widget never loads and while I do see the new comment badge count on posts, new comments aren't being highlighted.

**Comment by thexYeti** [comment]
just letting you know that the new comment highlighter works reliably on the outside thread, but doesn’t highlight the actual comment inside the thread. i think it’s supposed to do that, right? 

iphone 12 pro, iOS 17 with trollstore
![IMG_5235](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/assets/113392391/c669f7e6-1c7e-4468-8a83-7953a6b4c510)
, apollo 1.15.11

**Comment by Proaxel** [comment]
I forgot, the device I mentioned earlier is also jailbroken so I can just install the deb and not need to mess with sideloading.

But as it turns out I apparently can’t use the tweak at all on it because it requires iOS 15 or newer… is there anything that can be done about this?

![IMG_4168](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/assets/9545347/c858a5b5-de83-4f90-b6fc-d17dda5654c2)

**Comment by JeffreyCA** [comment]
> I forgot, the device I mentioned earlier is also jailbroken so I can just install the deb and not need to mess with sideloading.
> 
> But as it turns out I apparently can’t use the tweak at all on it because it requires iOS 15 or newer… is there anything that can be done about this?
> 
> ![IMG_4168](https://private-user-images.githubusercontent.com/9545347/288233824-c858a5b5-de83-4f90-b6fc-d17dda5654c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE4MzgzNjksIm5iZiI6MTcwMTgzODA2OSwicGF0aCI6Ii85NTQ1MzQ3LzI4ODIzMzgyNC1jODU4YTViNS1kZTgzLTRmOTAtYjZmYy1kMTdkZGE1NjU0YzIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDZUMDQ0NzQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGY0ZjlhY2U3NzQwOWQzMTZiMzViNGMyM2IwNjMxOGZhZDkyZDg3MDZlYTVlZTQ2NWJjMTNkNzc5OGU3YjMyMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.7myOKR4mFccje2RHkkl7oVHeuN3f6oPXCRi9JBklUHs)

Can you try [1.0.6b](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/releases/tag/v1.0.6b-experimental)? I lowered the iOS version requirement

**Comment by thexYeti** [comment]
> Does the Subreddit Weather feature work for you? It needs the com.apple.developer.weatherkit entitlement I believe.

apologies for the late reply @JeffreyCA

The subreddit weather appears to be not working, i checked a few big cities‘ subreddits. 
It does have the entitlement com.apple.developer.weatherkit, i included a screen for more info (sourced with appindex)

regarding the new comment highlighter screen you included, i did some more testing and it appears to be not working within the thread (like in the screen), only on the outside, whether in dark or light mode. 

I’m happy to help with testing, though i only have this device. (12 pro on ios 17.0 with trollstore, currently using apollo 1.15.11) 

cheers and thanks 

![IMG_5245](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/assets/113392391/8ccb7bac-1e6e-4b6d-943e-211a5b85f613)
![IMG_5246](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/assets/113392391/13162a3d-b84d-4949-9ff6-125122a0064f)

**Comment by JeffreyCA** [comment]
Thanks for confirming! So don't see a loading spinner like the following or after toggling off and on? 

For reference this is what I get:

![image](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/assets/9157833/93dce592-7e81-4640-9084-af61b6796a17)

**Comment by thexYeti** [comment]
no i dont, tested toggling on/off a few times, also tested the new 1.0.7exp version. 
not that i personally think the sub weather is a very important feature lol. 
if you need logs or anything, let me know

**Comment by Fofer** [comment]
> I don’t think it’s just entitlements. 1.15.11 should have all ultra features unlocked, **but even the Pixel Pals don’t work.**

Interestingly enough, "Cat" and "Dog" Pixel Pals work for me. It's only when trying to select any of the other Pixel Pals (Hedgehog, Axolotl, Fox, Otter, etc.) that the app crashes immediately. This is by no means a big deal, I don't really care for Pixel Pals anyway, just sharing what I discovered when tapping around out of curiosity.  I am so, so happy that Apollo is resurrected on my iPhone!

**Comment by cameronwilliams** [comment]
> > I don’t think it’s just entitlements. 1.15.11 should have all ultra features unlocked, **but even the Pixel Pals don’t work.**
> 
> Interestingly enough, the Cat, Dog, and Artificial Superintelligence Pixel Pals work for me. It's only when trying to select any of the other Pixel Pals (Hedgehog, Axolotl, Fox, Otter, etc.) that the app crashes immediately. This is by no means a big deal, I don't really care for Pixel Pals anyway, just sharing what I discovered when tapping around out of curiosity. I am so, so happy that Apollo is resurrected on my iPhone!

The Cat and Dog work because they’re free. The rest are Ultra features and don’t work. I believe the issue is caused by it thinking it’s the TestFlight version. For the regular version, they’re unlocked, for the TestFlight version, they’re unlockable for free, but only with the IAP entitlement.

**Comment by Fofer** [comment]
> The Cat and Dog work because they’re free.

Understood. Just saying though that the “artificial superintelligence” Pixel Pal (which was also part of Ultra) works too.

**Comment by mustafacakirogludev** [comment]
@JeffreyCA Hey, I have a question regarding apollo ultra features. I believe we can add notification support to Apollo for the users who have an apple developer account but I might need some help. Normally if you create an identifier and a provision profile with push notification entitlements and give the correct bundleId while signing(in this case reddit's bundle id), you can get push notifications for sideloaded apps. I am using a modded Discord client and using this method I am getting my discord notifications. The problem is, apollo crashes when you try to enable "inbox notifications" toggle in settings, I believe  the ultra subscription check causes the crash. I'm pretty sure if I can enable that toggle, I'll get push notifications. Do you think you can help me with that in a future release?

**Comment by samuelawachie** [comment]
Thanks for this; it’s very impressive. 
Is there any timeline for the new comments highlighter to show inside posts? That’s the only thing left to make this total perfection imo. I know devs don’t like when people ask for ETA but is it being worked on?

**Comment by JeffreyCA** [comment]
> Thanks for this; it’s very impressive. Is there any timeline for the new comments highlighter to show inside posts? That’s the only thing left to make this total perfection imo. I know devs don’t like when people ask for ETA but is it being worked on?

Unfortunately not... if anyone has ideas on how to fix please comment or submit a PR

**Comment by samuelawachie** [comment]
> > Thanks for this; it’s very impressive. Is there any timeline for the new comments highlighter to show inside posts? That’s the only thing left to make this total perfection imo. I know devs don’t like when people ask for ETA but is it being worked on?
> 
> Unfortunately not... if anyone has ideas on how to fix please comment or submit a PR

The thing is, links in comments threads that point to another comment in a comment thread work. When tapped, it takes you to the comment thread reference and highlights the comment that was linked, with the entire comment thread still there as context. 
I would’ve assumed the code to fix this would be something similar where it opens the context, and highlights new rather than linked comment. Or it could sort by new and then highlights the comments filtered out by date just using the code for linked comments. 
The reason I say this is that those linked comments work so similarly to the behavior we wish to implement.

**Comment by deeerooo** [comment]
> > I don’t think it’s just entitlements. 1.15.11 should have all ultra features unlocked, **but even the Pixel Pals don’t work.**
> 
> 
> 
> Interestingly enough, the Cat, Dog, and Artificial Superintelligence Pixel Pals work for me. It's only when trying to select any of the other Pixel Pals (Hedgehog, Axolotl, Fox, Otter, etc.) that the app crashes immediately. This is by no means a big deal, I don't really care for Pixel Pals anyway, just sharing what I discovered when tapping around out of curiosity.  I am so, so happy that Apollo is resurrected on my iPhone!
> 
> 

I just happened to be messing around with preferences in Filza, and if you have access, you can change to any pixel pal.

It's in group. com.christianselig.Apollo.plist,

"ActivePixelPal" . Just use the lower case version of the pal. Don't really care for them either but cool to have back!

**Comment by JeffreyCA** [comment]
Thanks for your patience everyone, a working version of New Comments Highlighter is now available in [v1.1.1](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/releases/tag/v1.1.1) (#32)!

Please report any issues/bugs you encounter - I had to make a semi-custom implementation so it may not behave exactly as the original one.

**Comment by samuelawachie** [comment]
> Thanks for your patience everyone, a working version of New Comments Highlighter is now available in [v1.1.1](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/releases/tag/v1.1.1) (#32)!
> 
> 
> 
> Please report any issues/bugs you encounter - I had to make a semi-custom implementation so it may not behave exactly as the original one.
>
Way to go mate! Well done! I'm using it now and it's working beautifully so far. What's the FLEX option really? Is it for debugging the app?
I know I was nearly a nuisance asking for this feature, but umm I'm so glad it's here now.

**Comment by JeffreyCA** [comment]
Yes FLEX is for debugging (more info [here](https://github.com/FLEXTool/FLEX). It's usually used in tweaks for other apps as well

**Comment by thexYeti** [comment]
Works really well now, props for making it work and thanks!

**Comment by JeffreyCA** [comment]
Excited to finally release [v2](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/releases/tag/v2.0.0) with saved categories, unlocked icons & Pixel Pals and more, please give it try!

And all of this wouldn't have been possible without GitHub Copilot and Claude Code, which did most of the investigation and code generation.

**Comment by tangchr1** [comment]
V2 is working great. Thanks!!

**Comment by deeerooo** [comment]
> Excited to finally release [v2](https://github.com/JeffreyCA/Apollo-ImprovedCustomApi/releases/tag/v2.0.0) with saved categories, unlocked icons & Pixel Pals and more, please give it try!
> 
> And all of this wouldn't have been possible without GitHub Copilot and Claude Code, which did most of the investigation and code generation.

Did you not notice my reply about enabling pixels pals and icons from the p
List? They were already available when sideloaded, at least for me. Good job!
